### ISSUE TITLE
Make editor.action.marker.nextInFiles prioritize severe issues

### ISSUE THREAD

**Issue by tgeng** [microsoft/vscode#97643]
The command editor.action.goToTypeDefinition navigates to the next issue regardless of the severity. IMHO this makes the functionality less useful, especially if you have many extensions that give you all kinds of "help" and you likely only cares about compile issues 80% of the time.

On the other hand, IntelliJ only navigates to a warning when the entire file does not have any errors. Can we have a setting that tweaks this behavior?

**Comment by jrieken** [comment]
> The command editor.action.goToTypeDefinition navigates to the next issue regardless of the severity

No. It always goes to errors first, then warnings, then infos. Not sure what you are asking for

![May-13-2020 09-34-03](https://user-images.githubusercontent.com/1794099/81784319-f9182500-94fc-11ea-8c6e-2999b2a5a8d5.gif)

**Comment by tgeng** [comment]
Sorry for not providing enough details. I want complete mimicking the behavior of IntelliJ, which means two things
1. Only go to the next error/warning/hint/etc in the current file and do not jump to a different file.
2. When the list of error is exhausted, wrap around and go to the first *error* rather than the first *warning*. The warning list is only considered when the entire file has zero errors. That is,  with the screenshot you have, line 21 should never be in focus while user keep pressing F8 because there are more important places (errors) the user should care about first.

**Comment by tgeng** [comment]
Another point that I forgot to mention: It would be more useful if nextInFiles and prevInFiles could respect the current cursor position. Maybe adding new commands like "nextIssueInCurrentFileAfterCursor" and "prevIssueInCurrentFileBeforeCursor".
